pkg/collector/corechecks/snmp/internal/checkconfig/config.go

```diff
@@ -431,14 +439,33 @@ func NewCheckConfig(rawInstance integration.Data, rawInitConfig integration.Data
 	}
 
 	// SNMP connection configs
-	c.CommunityString = instance.CommunityString
-	c.User = instance.User
-	c.AuthProtocol = instance.AuthProtocol
-	c.AuthKey = instance.AuthKey
-	c.PrivProtocol = instance.PrivProtocol
-	c.PrivKey = instance.PrivKey
-	c.ContextName = instance.ContextName
+	if instance.CommunityString != "" {
+		c.CommunityString = instance.CommunityString
+	}
 
+	if instance.User != "" {
+		c.User = instance.User
+
+		if instance.AuthKey != "" {
+			c.AuthKey = instance.AuthKey
+			if instance.AuthProtocol == "" {
+				c.AuthProtocol = authProtocolMapping["usmHMACMD5AuthProtocol"]
+			} else {
+				c.AuthProtocol = instance.AuthProtocol
+			}
+		}
+
+		if instance.PrivKey != "" {
+			c.PrivKey = instance.PrivKey
+			if instance.PrivProtocol == "" {
+				c.PrivProtocol = privProtocolMapping["usmDESPrivProtocol"]
+			} else {
+				c.PrivProtocol = instance.PrivProtocol
+			}
+		}
+	}
+
+	c.ContextName = instance.ContextName
```

Comment:
Hey 👋, thanks for the PR.

About the processing of SNMP credentials (including auth key/proto, priv key/proto), I think it might be better to have those changes in session.go here: https://github.com/DataDog/datadog-agent/blob/14736d30acaad238612e27de363a37b837e5a8e3/pkg/collector/corechecks/snmp/internal/session/session.go#L89-L97

Maybe could be something like:

```
		if config.AuthKey != "" && config.AuthProtocol == "" {
			config.AuthProtocol = "md5"
		}
		if config.PrivKey != "" && config.PrivProtocol == "" {
			config.PrivProtocol = "sha"
		}
```